go/storage/mkvs/db/badger/badger.go

```diff
@@ -680,11 +745,19 @@ func (d *badgerNodeDB) NewBatch(oldRoot node.Root, version uint64, chunk bool) a
 	// thing to do would be to either crank up MaxTableSize or maybe split
 	// the transaction out.
 
+	var logBatch *badger.WriteBatch
+	if d.multipart {
+		// The node log is at a different version than the nodes themselves,
+		// which is awkward.
+		logBatch = d.db.NewWriteBatchAt(tsMetadata)
+	}
+
 	return &badgerBatch{
-		db:      d,
-		bat:     d.db.NewWriteBatchAt(versionToTs(version)),
-		oldRoot: oldRoot,
-		chunk:   chunk,
+		db:             d,
+		bat:            d.db.NewWriteBatchAt(versionToTs(version)),
+		multipartNodes: logBatch,
+		oldRoot:        oldRoot,
+		chunk:          chunk,
```

Comment:
Could we make sure that `chunk` must equal `d.multipart` otherwise batch operations error? It would be even better if `NewBatch` could return an error and then we could fail early.